lib/arrow.cpp

```diff
@@ -1808,7 +1848,11 @@ namespace grnarrow {
       schema_builder_.Reset();
       schema_ = *schema;
 
-#if ARROW_VERSION_MAJOR >= 2
+#if ARROW_VERSION_MAJOR >= 3
+      auto option = arrow::ipc::IpcWriteOptions::Defaults();
+      option.emit_dictionary_deltas = true;
+      auto writer = arrow::ipc::MakeStreamWriter(&output_, schema_, option);
+#elif ARROW_VERSION_MAJOR >= 2
```

Comment:
Question:
When do we use an older version arrow although we specify the specific version array in a submodule ?
Can we delete these `ARROW_VERSION_MAJOR` checks ?

Reply:
We didn't use submoduled Apache Arrow automatically. We only find Apache Arrow installed in system:

* https://github.com/groonga/groonga/blob/master/CMakeLists.txt#L882
* https://github.com/groonga/groonga/blob/master/configure.ac#L948

We build submoduled Apache Arrow explicitly in MSVC CI:

* https://github.com/groonga/groonga/blob/master/.github/workflows/windows-msvc.yml#L129

We can remove these checks by updating required Apache Arrow versions in `CMakeLists.txt` and `configure.ac`.

Reply:
Thank you!
I have created a new PR to update arrow version.
#1265

Reply:
#1265 was merged